src/tasks/barfTasks.ts

```diff
@@ -264,76 +259,82 @@ const turns: AdventureAction[] = [
           Macro.externalIf(underwater, Macro.item($item`pulled green taffy`)).meatKill()
         ).abort()
       );
-      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+      return;
     },
     spendsTurn: () => !SourceTerminal.getDigitizeMonster()?.attributes.includes("FREE"),
     sobriety: Sobriety.EITHER,
+    completed: () => get("_sourceTerminalDigitizeMonsterCount") !== digitize,
   },
   {
     name: "Guaranteed Kramco",
-    available: () => kramcoGuaranteed(),
-    execute: () => {
-      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+    ready: () => kramcoGuaranteed(),
+    do: (): void => {
+      prep("free", {
+        requirements: new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }),
+      });
       adventureMacroAuto(drunkSafeWander("wanderer"), Macro.basicCombat());
-      return !kramcoGuaranteed();
     },
+    completed: () => !kramcoGuaranteed(),
     spendsTurn: false,
     sobriety: Sobriety.EITHER,
   },
   {
     name: "Void Monster",
-    available: () =>
+    ready: () =>
       have($item`cursed magnifying glass`) &&
       get("cursedMagnifyingGlassCount") === 13 &&
       get("_voidFreeFights") < 5,
-    execute: () => {
-      freeFightPrep(new Requirement([], { forceEquip: $items`cursed magnifying glass` }));
+    do: (): void => {
+      prep("free", {
+        requirements: new Requirement([], { forceEquip: $items`cursed magnifying glass` }),
+      });
       adventureMacroAuto(drunkSafeWander("wanderer"), Macro.basicCombat());
-      return get("cursedMagnifyingGlassCount") === 0;
     },
+    completed: () => get("cursedMagnifyingGlassCount") === 0,
     spendsTurn: false,
     sobriety: Sobriety.EITHER,
   },
   {
     name: "Envyfish Egg",
-    available: () =>
+    ready: () =>
       have($item`envyfish egg`) && get("envyfishMonster") === embezzler && !get("_envyfishEggUsed"),
-    execute: () => {
-      embezzlerPrep();
+    do: (): void => {
+      prep("meat");
       withMacro(Macro.meatKill(), () => use($item`envyfish egg`), true);
-      return get("_envyfishEggUsed");
     },
+    completed: () => get("_envyfishEggUsed"),
     spendsTurn: true,
     sobriety: Sobriety.EITHER,
   },
   {
     name: "Spit Acid",
-    available: () =>
+    ready: () =>
       have($item`Jurassic Parka`) &&
       !have($effect`Everything Looks Yellow`) &&
       romanticMonsterImpossible(),
-    execute: () => {
+    do: (): void => {
       const location = wanderWhere("yellow ray");
-      const familiar = freeFightFamiliar({ location });
-      useFamiliar(familiar);
-      freeFightOutfit(new Requirement([], { forceEquip: $items`Jurassic Parka` }));
+      prep("free", {
+        location,
+        requirements: new Requirement([], { forceEquip: $items`Jurassic Parka` }),
```

Comment:
almost every `Requirement` passed in the prep function is a singleton requirement of a forceEquip; there might be a nicer way to set it up.

We do occasionally see nicer ones, though.